packages/tx-construction/test/tx-builder/TxBuilder.test.ts

```diff
@@ -349,6 +405,25 @@ describe('GenericTxBuilder', () => {
       it('legit output with valid with address and coin', async () => {
         await expect(txBuilder.buildOutput().address(address).coin(output1Coin).build()).resolves.toBeTruthy();
       });
+
+      it('resolves handle to address', async () => {
+        const txOut = await txBuilder.buildOutput().handle('alice').coin(output1Coin).build();
+
+        expect(txOut.handle).toBe(resolvedHandle);
+        expect(txOut.address).toBe(resolvedHandle.resolvedAddresses.cardano);
+      });
+
+      it('rejects with an error when a handle is not found', async () => {
+        await expect(
+          txBuilderWithNullHandles.buildOutput().handle('alice').coin(output1Coin).build()
+        ).rejects.toThrowError(OutputValidationMissingRequiredError);
```

Comment:
Maybe let's rename this to better communicate what's actually happening
```suggestion
        ).rejects.toThrowError(HandleNotFoundError);
```